spec/integration/admin_api/kong_routes_spec.lua

```diff
@@ -83,4 +84,42 @@ describe("Admin API", function()
       assert.truthy(body.server.total_requests)
     end)
   end)
+
+  describe("Request size", function()
+    it("should properly hanlde big POST bodies < 10MB", function()
+      local response, status = http_client.post(spec_helper.API_URL.."/apis", { request_path = "hello.com", upstream_url = "http://mockbin.org" })
+      assert.equal(201, status)
+      local api_id = json.decode(response).id
+      assert.truthy(api_id)
+
+      local big_value = {}
+      for i=1, 1000 do
+        table.insert(big_value, "204.48.16.0")
+      end
+      local big_value_str = table.concat(big_value, ",")
```

Comment:
It would have been cleaner to use string.rep until a certain body size is needed. That is what the Galileo buffer tests are doing for example.

-- 
Thibault Charbonnier
Sent with Airmail

On November 9, 2015 at 2:43:08 PM, Marco Palladino (notifications@github.com) wrote:

In spec/integration/admin_api/kong_routes_spec.lua:

> @@ -83,4 +84,42 @@ describe("Admin API", function()
>        assert.truthy(body.server.total_requests)
>      end)
>    end)
> +
> +  describe("Request size", function()
> +    it("should properly hanlde big POST bodies < 10MB", function()
> +      local response, status = http_client.post(spec_helper.API_URL.."/apis", { request_path = "hello.com", upstream_url = "http://mockbin.org" })
> +      assert.equal(201, status)
> +      local api_id = json.decode(response).id
> +      assert.truthy(api_id)
> +
> +      local big_value = {}
> +      for i=1, 1000 do
> +        table.insert(big_value, "204.48.16.0")
> +      end
> +      local big_value_str = table.concat(big_value, ",")
@thibaultCha I opted for this way instead to build a JSON value in memory

—
Reply to this email directly or view it on GitHub.